### DISCUSSION TITLE
Why `dict()` does not produce a list representation of `NamedTuple`?

### DISCUSSION THREAD

**Discussion by atugushev** [pydantic/pydantic#4381] [Question] [answered]
`BaseModel.dict()` doesn't produce lists for `NamedTuple`-s for some reason (see example below).  Moreover, this looks like a deliberate decision (note `is_namedtuple` check): https://github.com/pydantic/pydantic/blob/4e7878932ca5d522e4df16f9821b1cdd324344cf/pydantic/main.py#L786

I've tried `json()` and `parse_raw()` they work as expected . I wonder, is there a history behind this or am I missing something in docs?

```python
import pydantic, typing


class FullName(typing.NamedTuple):
    first_name: str
    last_name: str


class User(pydantic.BaseModel):
    id: int
    full_name: FullName


user = User(id=1, full_name=["Albert", "Einstein"])

# Good
print('parse_raw:', repr(User.parse_raw('{"id": 1, "full_name": ["Albert", "Einstein"]}')))
#> parse_raw: User(id=1, full_name=FullName(first_name='Albert', last_name='Einstein'))

# Good
print('json:', user.json())
#> json: {"id": 1, "full_name": ["Albert", "Einstein"]}

# No good
print('dict:', user.dict())
#> dict: {'id': 1, 'full_name': FullName(first_name='Albert', last_name='Einstein')}

# Expecting
#> dict: {'id': 1, 'full_name': ['Albert', 'Einstein']}

```

**Comment by samuelcolvin** [comment, marked as answer]
I think you're conflating a few things:


Obviously there's no direct representation of named-tuples in JSON, so we have to:
1. Allow conversion from array's in JSON to named-tuples
2. Convert named-tuples to arrays when dumping to JSON

The case of `dict()` is different, we convert pydantic models to dicts recursively, but we don't do any other conversion. This might change in V2 where we're [redesigning data serialisation](https://pydantic-docs.helpmanual.io/blog/pydantic-v2/#improvements-to-dumpingserializationexport)
